integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/rdd/SecondaryIndexCreator.scala

```diff
@@ -385,7 +392,26 @@ object SecondaryIndexCreator {
         val rebuiltSegments = SecondaryIndexUtil
           .mergeDataFilesSISegments(secondaryIndexModel.segmentIdToLoadStartTimeMapping,
             indexCarbonTable,
-            loadMetadataDetails.toList.asJava, carbonLoadModelForMergeDataFiles)(sc)
+            loadMetadataDetail.toList.asJava, carbonLoadModelForMergeDataFiles)(sc)
+        if (isInsertOverwrite) {
+          var segmentList = new ListBuffer[String]()
+          for (loadMetadata <- loadMetadataDetails) {
+            if (loadMetadata.getSegmentStatus != SegmentStatus.INSERT_OVERWRITE_IN_PROGRESS) {
+              segmentList += loadMetadata.getLoadName
+            }
+          }
+          if (segmentList.nonEmpty) {
```

Comment:
please change the status to marked for delete after making the new insert overwritten segment to success. else we may get issues of reliability in concurrent scenarios

Reply:
done